src/Optimization/hiopNlpFormulation.cpp

```diff
@@ -282,9 +282,9 @@ bool hiopNlpFormulation::finalizeInitialization()
 
   dFixedVarsTol = fixedVarTol;
   
-  long long nfixed_vars=nfixed_vars_local;
+  int_type nfixed_vars=nfixed_vars_local;
 #ifdef HIOP_USE_MPI
-  int ierr = MPI_Allreduce(&nfixed_vars_local, &nfixed_vars, 1, MPI_LONG_LONG, MPI_SUM, comm); 
+  int ierr = MPI_Allreduce(&nfixed_vars_local, &nfixed_vars, 1, MPI_INT, MPI_SUM, comm); 
```

Comment:
Probably good idea to implement index traits class to manage issues like this. That should be straightforward.